src/components/Calendar/index.js

```diff
@@ -52,30 +171,119 @@ class CalendarComponent extends Component {
         });
     }
 
+    updateFocusedDate(value) {
+        this.setState({
+            focusedDate: value,
+        });
+    }
+
     nextMonth() {
+        const newMonth = addMonths(this.state.currentMonth, 1);
+        const { value } = this.props;
+        const focusedDate =
+            isSameMonth(value, newMonth) && isSameYear(value, newMonth)
+                ? value
+                : getFirstDayMonth(newMonth);
+
         this.setState({
-            currentMonth: addMonths(this.state.currentMonth, 1),
+            focusedDate,
+            currentMonth: newMonth,
         });
     }
 
     previousMonth() {
+        const newMonth = addMonths(this.state.currentMonth, -1);
+        const { value } = this.props;
+        const focusedDate =
+            isSameMonth(value, newMonth) && isSameYear(value, newMonth)
+                ? value
+                : getFirstDayMonth(newMonth);
+
         this.setState({
-            currentMonth: addMonths(this.state.currentMonth, -1),
+            focusedDate,
+            currentMonth: newMonth,
         });
     }
 
     handleYearChange(event) {
         const year = +event.target.value;
         const newMonth = new Date(this.state.currentMonth);
         newMonth.setFullYear(year);
+
+        const { value } = this.props;
+        const focusedDate =
+            isSameMonth(value, newMonth) && isSameYear(value, newMonth)
+                ? value
+                : getFirstDayMonth(newMonth);
```

Comment:
this is repeated three times so use a helper for this (getNextFocusedDate) and test it